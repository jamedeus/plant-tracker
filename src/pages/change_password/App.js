import React, { useState, useMemo, useRef } from 'react';
import { useTheme } from 'src/context/ThemeContext';
import Navbar from 'src/components/Navbar';
import Cookies from 'js-cookie';
import clsx from 'clsx';

function App() {
    const { ToggleThemeOption } = useTheme();

    const formRef = useRef(null);
    const [oldPassword, setOldPassword] = useState('');
    const [newPassword1, setNewPassword1] = useState('');
    const [newPassword2, setNewPassword2] = useState('');
    const [oldPasswordIncorrect, setOldPasswordIncorrect] = useState(false);
    const [newPasswordError, setNewPasswordError] = useState(false);

    const submit = async (e) => {
        e.preventDefault();
        // Post old and new password to backend
        const response = await fetch("/accounts/change_password/submit/", {
            method: "POST",
            headers: {
                "Content-Type": "application/x-www-form-urlencoded",
                "X-CSRFToken": Cookies.get('csrftoken')
            },
            credentials: "include",
            body: new URLSearchParams({
                old_password: oldPassword,
                new_password1: newPassword1,
                new_password2: newPassword2,
            }),
        });
        // Redirect to overview if logged in successfully
        if (response.ok) {
            window.location.href = '/';
        // Show correct error message if login failed
        } else {
            const errors = await response.json();
            if (errors.errors.old_password) {
                setOldPasswordIncorrect(true);
            }
            if (errors.errors.new_password2) {
                setNewPasswordError(errors.errors.new_password2[0]);
            }
        }
    };

    const DropdownMenuOptions = useMemo(() => (
        <>
            <li><a onClick={() => window.location.href = "/"}>
                Overview
            </a></li>
            <ToggleThemeOption />
        </>
    ), [ToggleThemeOption]);

    return (
        <div className="container flex flex-col h-screen mx-auto items-center">
            <Navbar
                menuOptions={DropdownMenuOptions}
                title="Change Password"
            />

            <form ref={formRef} className="flex flex-col gap-4 mt-[15vh]">
                <label className="form-control w-full">
                    <div className="label">
                        <span className="label-text">Old password</span>
                    </div>
                    <input
                        name="old_password"
                        type="password"
                        className={clsx(
                            "input w-full input-bordered",
                            oldPasswordIncorrect && "input-error"
                        )}
                        value={oldPassword}
                        onInput={() => setOldPasswordIncorrect(false)}
                        onChange={(e) => setOldPassword(e.target.value)}
                    />
                </label>
                {oldPasswordIncorrect && (
                    <span className="text-error">
                        Old password incorrect
                    </span>
                )}
                <label className="form-control w-full relative">
                    <div className="label">
                        <span className="label-text">New password</span>
                    </div>
                    <input
                        name="new_password1"
                        type="password"
                        className={clsx(
                            "input w-full input-bordered",
                            newPasswordError && "input-error"
                        )}
                        value={newPassword1}
                        onChange={(e) => setNewPassword1(e.target.value)}
                    />
                </label>
                <label className="form-control w-full">
                    <div className="label">
                        <span className="label-text">Confirm new password</span>
                    </div>
                    <input
                        name="new_password2"
                        type="password"
                        className={clsx(
                            "input w-full input-bordered",
                            newPasswordError && "input-error"
                        )}
                        value={newPassword2}
                        onInput={() => setNewPasswordError(false)}
                        onChange={(e) => setNewPassword2(e.target.value)}
                    />
                </label>
                {newPasswordError && (
                    <span className="text-error">
                        {newPasswordError}
                    </span>
                )}

                <button
                    className="btn btn-success mt-6"
                    onClick={(e) => submit(e)}
                >
                    Change password
                </button>
            </form>
        </div>
    );
}

export default App;
