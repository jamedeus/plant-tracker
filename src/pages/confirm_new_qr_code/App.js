import React, { useEffect, useMemo } from 'react';
import PropTypes from 'prop-types';
import { sendPostRequest, parseDomContext } from 'src/util';
import Navbar from 'src/components/Navbar';
import NavbarDropdownOptions from 'src/components/NavbarDropdownOptions';
import PlantDetails from 'src/components/PlantDetails.js';
import GroupDetails from 'src/components/GroupDetails.js';
import { openErrorModal } from 'src/components/ErrorModal';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faXmark, faCheck } from '@fortawesome/free-solid-svg-icons';

const Layout = ({ type, thumbnailUrl, handleConfirm }) => {
    return (
        <div className="flex flex-col gap-8 text-center my-auto">
            <p className="text-lg font-bold">
                Is this the new QR code for your {type}?
            </p>
            {thumbnailUrl && (
                <div className="mx-auto p-4 bg-base-200 rounded-3xl">
                    <img
                        className="max-h-[50vh] rounded-xl object-contain"
                        src={thumbnailUrl}
                    />
                </div>
            )}

            {/* Confirm/cancel buttons */}
            <div className="flex gap-4 mx-auto mb-8">
                <button
                    className="btn btn-error btn-square text-white"
                    onClick={() => window.location.href = "/"}
                    title="Don't change QR code"
                >
                    <FontAwesomeIcon className="size-6" icon={faXmark} />
                </button>
                <button
                    className="btn btn-success btn-square text-white"
                    onClick={handleConfirm}
                    title="Change QR code"
                >
                    <FontAwesomeIcon className="size-6" icon={faCheck} />
                </button>
            </div>
        </div>
    );
};

Layout.propTypes = {
    type: PropTypes.oneOf([
        'plant',
        'group'
    ]).isRequired,
    thumbnailUrl: PropTypes.string,
    handleConfirm: PropTypes.func.isRequired
};

function App() {
    // Load UUIDs and plant/group state from django template context
    const type = parseDomContext("type");
    const instance = parseDomContext("instance");
    const newUuid = parseDomContext("new_uuid");

    // Reload if user navigates to page by pressing back button (change QR code
    // request likely expired, replace with register or manage page)
    useEffect(() => {
        const handleBackButton = async (event) => {
            if (event.persisted) {
                window.location.reload();
            }
        };

        // Add listener on mount, remove on unmount
        window.addEventListener('pageshow', handleBackButton);
        return () => {
            window.removeEventListener('pageshow', handleBackButton);
        };
    }, []);

    const handleConfirm = async () => {
        const response = await sendPostRequest(
            '/change_uuid',
            {
                uuid: instance.uuid,
                new_id: newUuid
            }
        );
        // Reload page if changed successfully
        if (response.ok) {
            window.location.reload();
        } else {
            const error = await response.json();
            openErrorModal(JSON.stringify(error));
        }
    };

    // Plant/group details card shown when title is clicked
    const DetailsDropdown = useMemo(() => (
        <div className="details-card">
            <div className="card-body">
                {type === 'plant' ? (
                    <PlantDetails
                        species={instance.species}
                        pot_size={instance.pot_size}
                        description={instance.description}
                    />
                ) : (
                    <GroupDetails
                        location={instance.location}
                        description={instance.description}
                    />
                )}
            </div>
        </div>
    ), []);

    return (
        <div className="container flex flex-col mx-auto h-screen">
            <Navbar
                menuOptions={<NavbarDropdownOptions />}
                title={instance.display_name}
                titleOptions={DetailsDropdown}
            />
            <Layout
                type={type}
                thumbnailUrl={instance.thumbnail}
                handleConfirm={handleConfirm}
            />
        </div>
    );
}

export default App;
