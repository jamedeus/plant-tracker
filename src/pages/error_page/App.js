import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Navbar from 'src/components/Navbar';
import NavbarDropdownOptions from 'src/components/NavbarDropdownOptions';

function App({ errorMessage }) {
    return (
        <div
            className="container flex flex-col full-screen mx-auto items-center gap-16"
            data-testid="error-page"
        >
            <Navbar
                menuOptions={<NavbarDropdownOptions />}
                title="Error"
                showScanButton={false}
            />

            <div className="flex flex-col text-center text-lg mt-[15vh]">
                {errorMessage}
            </div>

            <Link to="/" className='btn btn-accent' discover="none">
                Go to Overview
            </Link>
        </div>
    );
}

export default App;

App.propTypes = {
    errorMessage: PropTypes.string.isRequired
};
