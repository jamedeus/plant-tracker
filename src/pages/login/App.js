import React, { useState, useMemo, useRef } from 'react';
import { useTheme } from 'src/context/ThemeContext';
import Navbar from 'src/components/Navbar';
import Cookies from 'js-cookie';
import clsx from 'clsx';

function App() {
    const { ToggleThemeOption } = useTheme();

    const formRef = useRef(null);
    const [username, setUsername] = useState('');
    const [password, setPassword] = useState('');
    const [showError, setShowError] = useState(false);

    const submit = async (e) => {
        e.preventDefault();
        // Post FormData to backend
        const formData = new FormData(formRef.current);
        const response = await fetch('/accounts/login/', {
            method: 'POST',
            body: formData,
            headers: {
                'Accept': 'application/json, text/plain, */*',
                "X-CSRFToken": Cookies.get('csrftoken')
            }
        });
        // Redirect if logged in successfully
        if (response.ok) {
            // Redirect to url in querystring if present
            const params = new URL(window.location.href).searchParams;
            if (params.get('next')) {
                window.location.href = params.get('next');
            // Redirect to overview if no querystring
            } else {
                window.location.href = '/';
            }
        // Show error text if login failed
        } else {
            setShowError(true);
        }
    };

    const DropdownMenuOptions = useMemo(() => (
        <>
            <li><a onClick={() => window.location.href = "/"}>
                Overview
            </a></li>
            <ToggleThemeOption />
        </>
    ), [ToggleThemeOption]);

    return (
        <div className="container flex flex-col h-screen mx-auto items-center">
            <Navbar
                menuOptions={DropdownMenuOptions}
                title="Login"
            />

            <form ref={formRef} className="flex flex-col gap-4 mt-[15vh]">
                <label className="form-control w-full">
                    <div className="label">
                        <span className="label-text">Username</span>
                    </div>
                    <input
                        name="username"
                        type="text"
                        autoCapitalize="off"
                        className={clsx(
                            "input w-full input-bordered",
                            showError && "input-error"
                        )}
                        value={username}
                        onInput={() => setShowError(false)}
                        onChange={e => setUsername(e.target.value)}
                    />
                </label>
                <label className="form-control w-full relative">
                    <div className="label">
                        <span className="label-text">Password</span>
                    </div>
                    <input
                        name="password"
                        type="password"
                        className={clsx(
                            "input w-full input-bordered",
                            showError && "input-error"
                        )}
                        value={password}
                        onInput={() => setShowError(false)}
                        onChange={e => setPassword(e.target.value)}
                    />
                </label>

                <button
                    className="btn btn-success mt-6"
                    onClick={(e) => submit(e)}
                    disabled={username.length < 1 || password.length < 1}
                >
                    Login
                </button>

                {showError && (
                    <span className="text-error text-center">
                        Invalid username or password
                    </span>
                )}
            </form>
        </div>
    );
}

export default App;
