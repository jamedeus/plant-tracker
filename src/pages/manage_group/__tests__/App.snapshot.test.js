import createMockContext from 'src/testUtils/createMockContext';
import App from '../App';
import { PageWrapper } from 'src/index';
import { mockContext } from './mockContext';

describe('App', () => {
    it('matches snapshot', () => {
        // Create mock state objects
        createMockContext('group', mockContext.group);
        createMockContext('details', mockContext.details);
        createMockContext('options', mockContext.options);

        // Render App, confirm matches snapshot
        const component = render(
            <PageWrapper>
                <App />
            </PageWrapper>
        );
        expect(component).toMatchSnapshot();
    });
});
