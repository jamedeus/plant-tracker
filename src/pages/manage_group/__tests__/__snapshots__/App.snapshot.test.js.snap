// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div>
  <div
    class="container flex flex-col items-center mx-auto mb-28"
  >
    <div
      class="navbar bg-base-100 fixed top-0 z-99"
    >
      <div
        class="dropdown justify-start min-w-12"
      >
        <div
          aria-label="Navigation menu"
          class="btn btn-ghost btn-circle size-12"
          role="button"
          tabindex="0"
        >
          <svg
            class="size-5"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 6h16M4 12h16M4 18h7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
        <ul
          class="dropdown-content menu z-90 shadow-sm bg-base-300 rounded-box mt-3"
          tabindex="0"
        >
          <li>
            <a
              href="/"
            >
              Overview
            </a>
          </li>
          <li>
            <a
              href="/accounts/profile/"
            >
              User profile
            </a>
          </li>
          <li>
            <a>
              Change QR code
            </a>
          </li>
          <li>
            <button>
              Light mode
            </button>
          </li>
        </ul>
      </div>
      <div
        class="mx-auto shrink min-w-0"
      >
        <div
          class="dropdown dropdown-center w-full"
        >
          <a
            class="w-full px-0 text-nowrap font-semibold btn btn-ghost border-0"
            role="button"
            tabindex="0"
          >
            <span
              class="px-4 leading-[2.75rem] truncate"
              style="font-size: 14px;"
            >
              Test group
            </span>
          </a>
          <div
            class="dropdown-content z-1 flex"
            tabindex="0"
          >
            <div
              class="details-card"
            >
              <div
                class="card-body text-sm"
              >
                <p
                  class="flex"
                >
                  <span
                    class="font-semibold break-keep mr-4"
                  >
                    Location:
                  </span>
                  <span
                    class="ml-auto text-right line-clamp-1 break-all"
                  >
                    Middle shelf
                  </span>
                </p>
                <button
                  class="btn h-8 mt-4"
                >
                  Edit
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="justify-end min-w-12"
      >
        <button
          aria-label="Open QR scanner"
          class="btn btn-ghost btn-circle size-12"
          title="Open QR scanner"
        >
          <svg
            class="size-6"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 7V5a2 2 0 0 1 2-2h2"
            />
            <path
              d="M17 3h2a2 2 0 0 1 2 2v2"
            />
            <path
              d="M21 17v2a2 2 0 0 1-2 2h-2"
            />
            <path
              d="M7 21H5a2 2 0 0 1-2-2v-2"
            />
            <circle
              cx="12"
              cy="12"
              r="3"
            />
            <path
              d="m16 16-1.9-1.9"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="flex flex-col items-center transition-[height] duration-300 h-[14.25rem]"
    >
      <div
        aria-orientation="horizontal"
        class="tab-group my-2 w-64"
        role="tablist"
      >
        <button
          aria-selected="true"
          class="tab-option whitespace-nowrap tab-option-selected"
          data-headlessui-state="selected"
          id="headlessui-tabs-tab-:r0:"
          role="tab"
          tabindex="0"
          type="button"
        >
          All plants
        </button>
        <button
          aria-selected="false"
          class="tab-option whitespace-nowrap"
          data-headlessui-state=""
          id="headlessui-tabs-tab-:r1:"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Select plants
        </button>
      </div>
      <input
        aria-label="Timestamp input"
        class="input my-2 text-center"
        step="1"
        type="datetime-local"
        value="2024-03-01T12:00:00"
      />
      <div
        class="flex mb-8"
      >
        <button
          class="btn btn-info m-2"
        >
          Water
        </button>
        <button
          class="btn btn-success m-2"
        >
          Fertilize
        </button>
      </div>
    </div>
    <div
      class="px-4 relative"
    >
      <div
        class="section"
      >
        <div
          class="section-header"
        >
          <div
            class="flex dropdown dropdown-center min-h-[3.75rem]"
          >
            <div
              class="section-title btn btn-ghost cursor-pointer"
              role="button"
              tabindex="0"
            >
              Plants (3)
            </div>
            <div
              class="dropdown-content z-50 mt-14"
            >
              <ul
                class="dropdown-content menu z-90 shadow-sm bg-base-300 rounded-box"
                tabindex="0"
              >
                <li>
                  <a
                    class="flex justify-center"
                    data-testid="add_plants_option"
                  >
                    Add
                  </a>
                </li>
                <li>
                  <a
                    class="flex justify-center"
                    data-testid="remove_plants_option"
                  >
                    Remove
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <div
            class="flex px-4 mb-4"
          >
            <div
              class="relative w-full"
            >
              <input
                class="input w-full text-center indent-[3.625rem] pr-[4.5rem]"
                placeholder="filter"
                type="text"
                value=""
              />
              <div
                class="absolute flex top-2 right-2"
              >
                <div
                  class="dropdown dropdown-end"
                >
                  <div
                    class="btn-close size-8"
                    role="button"
                    tabindex="0"
                    title="Sort menu"
                  >
                    <svg
                      aria-hidden="true"
                      class="size-5 m-auto"
                      data-slot="icon"
                      fill="currentColor"
                      viewBox="0 0 16 16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M13.78 10.47a.75.75 0 0 1 0 1.06l-2.25 2.25a.75.75 0 0 1-1.06 0l-2.25-2.25a.75.75 0 1 1 1.06-1.06l.97.97V5.75a.75.75 0 0 1 1.5 0v5.69l.97-.97a.75.75 0 0 1 1.06 0ZM2.22 5.53a.75.75 0 0 1 0-1.06l2.25-2.25a.75.75 0 0 1 1.06 0l2.25 2.25a.75.75 0 0 1-1.06 1.06l-.97-.97v5.69a.75.75 0 0 1-1.5 0V4.56l-.97.97a.75.75 0 0 1-1.06 0Z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </div>
                  <ul
                    class="dropdown-content menu z-90 shadow-sm bg-base-300 rounded-box mt-2"
                    tabindex="0"
                  >
                    <li>
                      <a
                        class="flex justify-end"
                      >
                        <svg
                          class="fa-arrow-down-long mr-2"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 384 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M169.4 502.6c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 402.7 224 32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 370.7L86.6 329.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128z"
                          />
                        </svg>
                        Added
                      </a>
                    </li>
                    <li>
                      <a
                        class="flex justify-end"
                      >
                        Name
                      </a>
                    </li>
                    <li>
                      <a
                        class="flex justify-end"
                      >
                        Species
                      </a>
                    </li>
                    <li>
                      <a
                        class="flex justify-end"
                      >
                        Watered
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="section-content"
        >
          <form
            class="flex flex-col gap-4"
          >
            <div
              class="flex relative"
            >
              <div
                class="absolute flex h-full z-0"
              >
                <input
                  aria-label="Select Test Plant"
                  class="radio my-auto"
                  id=":r2:-0640ec3b-1bed-4b15-a078-d6e7ec66be12"
                  name="0640ec3b-1bed-4b15-a078-d6e7ec66be12"
                  tabindex="-1"
                  type="checkbox"
                />
              </div>
              <div
                class="w-full overflow-hidden transition-all duration-300 ml-0"
              >
                <a
                  aria-label="Go to Test Plant page"
                  class="collapse cursor-pointer group rounded-2xl bg-neutral text-neutral-content"
                  href="/manage/0640ec3b-1bed-4b15-a078-d6e7ec66be12"
                >
                  <input
                    class="hidden pointer-events-none"
                    id=":r3:"
                    type="checkbox"
                  />
                  <div
                    class="collapse-title min-size-0"
                  >
                    <div
                      class="card card-side relative h-24"
                    >
                      <div
                        class="card-body max-w-full my-auto text-center"
                      >
                        <h2
                          class="card-title line-clamp-1 break-words text-center px-8"
                        >
                          Test Plant
                        </h2>
                        <span
                          class="line-clamp-1"
                          title="04:45 AM - February 29, 2024"
                        >
                          <svg
                            class="fa-inline mr-2 text-info"
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 384 512"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M192 512C86 512 0 426 0 320C0 228.8 130.2 57.7 166.6 11.7C172.6 4.2 181.5 0 191.1 0l1.8 0c9.6 0 18.5 4.2 24.5 11.7C253.8 57.7 384 228.8 384 320c0 106-86 192-192 192zM96 336c0-8.8-7.2-16-16-16s-16 7.2-16 16c0 61.9 50.1 112 112 112c8.8 0 16-7.2 16-16s-7.2-16-16-16c-44.2 0-80-35.8-80-80z"
                            />
                          </svg>
                          Yesterday
                        </span>
                      </div>
                      <label
                        aria-label="Show or hide details"
                        class="btn-close absolute right-2 top-8 z-40"
                        for=":r3:"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class="min-size-8 transition-transform duration-200 rotate-0 group-has-checked:rotate-180"
                          data-slot="icon"
                          fill="currentColor"
                          viewBox="0 0 16 16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M4.22 6.22a.75.75 0 0 1 1.06 0L8 8.94l2.72-2.72a.75.75 0 1 1 1.06 1.06l-3.25 3.25a.75.75 0 0 1-1.06 0L4.22 7.28a.75.75 0 0 1 0-1.06Z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </label>
                    </div>
                  </div>
                  <div
                    class="collapse-content min-w-0"
                  >
                    <div
                      class="pt-4"
                    >
                      <p
                        class="flex"
                      >
                        <span
                          class="font-semibold break-keep mr-4"
                        >
                          Species:
                        </span>
                        <span
                          class="ml-auto text-right line-clamp-1 break-all"
                        >
                          Calathea
                        </span>
                      </p>
                      <p
                        class="flex"
                      >
                        <span
                          class="font-semibold break-keep mr-4"
                        >
                          Pot size:
                        </span>
                        <span
                          class="ml-auto text-right line-clamp-1 break-all"
                        >
                          4
                        </span>
                      </p>
                      <div
                        class="text-center"
                      >
                        <p
                          class="font-semibold mt-3"
                        >
                          Description:
                        </p>
                        <p
                          class="text-sm line-clamp-6 break-anywhere"
                        >
                          This is a plant with a long description with
                        </p>
                      </div>
                    </div>
                  </div>
                </a>
              </div>
              <label
                class="cursor-pointer absolute z-20"
                for=":r2:-0640ec3b-1bed-4b15-a078-d6e7ec66be12"
              />
            </div>
            <div
              class="flex relative"
            >
              <div
                class="absolute flex h-full z-0"
              >
                <input
                  aria-label="Select node"
                  class="radio my-auto"
                  id=":r2:-19f65fa0-1c75-4cba-b590-0c9b5b315fcc"
                  name="19f65fa0-1c75-4cba-b590-0c9b5b315fcc"
                  tabindex="-1"
                  type="checkbox"
                />
              </div>
              <div
                class="w-full overflow-hidden transition-all duration-300 ml-0"
              >
                <a
                  aria-label="Go to Unnamed Spider Plant page"
                  class="collapse cursor-pointer group rounded-2xl bg-neutral text-neutral-content grayscale"
                  href="/manage/19f65fa0-1c75-4cba-b590-0c9b5b315fcc"
                >
                  <input
                    class="hidden pointer-events-none"
                    id=":r4:"
                    type="checkbox"
                  />
                  <div
                    class="collapse-title min-size-0"
                  >
                    <div
                      class="card card-side relative h-24"
                    >
                      <figure
                        class="h-24 w-20 min-size-20"
                      >
                        <img
                          alt="Unnamed Spider Plant photo"
                          class="size-full object-cover"
                          draggable="false"
                          loading="lazy"
                          src="/media/thumbnails/photo_thumb.webp"
                        />
                      </figure>
                      <div
                        class="card-body max-w-full my-auto text-start"
                      >
                        <h2
                          class="card-title line-clamp-1 break-words pr-8"
                        >
                          Unnamed Spider Plant
                        </h2>
                        <span
                          class="line-clamp-1"
                          title="04:45 AM - February 29, 2024"
                        >
                          <svg
                            class="fa-inline mr-2 text-info"
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 384 512"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M192 512C86 512 0 426 0 320C0 228.8 130.2 57.7 166.6 11.7C172.6 4.2 181.5 0 191.1 0l1.8 0c9.6 0 18.5 4.2 24.5 11.7C253.8 57.7 384 228.8 384 320c0 106-86 192-192 192zM96 336c0-8.8-7.2-16-16-16s-16 7.2-16 16c0 61.9 50.1 112 112 112c8.8 0 16-7.2 16-16s-7.2-16-16-16c-44.2 0-80-35.8-80-80z"
                            />
                          </svg>
                          Yesterday
                        </span>
                      </div>
                      <label
                        aria-label="Show or hide details"
                        class="btn-close absolute right-2 top-8 z-40"
                        for=":r4:"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class="min-size-8 transition-transform duration-200 rotate-0 group-has-checked:rotate-180"
                          data-slot="icon"
                          fill="currentColor"
                          viewBox="0 0 16 16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M4.22 6.22a.75.75 0 0 1 1.06 0L8 8.94l2.72-2.72a.75.75 0 1 1 1.06 1.06l-3.25 3.25a.75.75 0 0 1-1.06 0L4.22 7.28a.75.75 0 0 1 0-1.06Z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </label>
                    </div>
                  </div>
                  <div
                    class="collapse-content min-w-0"
                  >
                    <div
                      class="pt-4"
                    >
                      <p
                        class="flex"
                      >
                        <span
                          class="font-semibold break-keep mr-4"
                        >
                          Species:
                        </span>
                        <span
                          class="ml-auto text-right line-clamp-1 break-all"
                        >
                          Spider Plant
                        </span>
                      </p>
                      <p
                        class="flex"
                      >
                        <span
                          class="font-semibold break-keep mr-4"
                        >
                          Pot size:
                        </span>
                        <span
                          class="ml-auto text-right line-clamp-1 break-all"
                        >
                          2
                        </span>
                      </p>
                    </div>
                  </div>
                </a>
              </div>
              <label
                class="cursor-pointer absolute z-20"
                for=":r2:-19f65fa0-1c75-4cba-b590-0c9b5b315fcc"
              />
            </div>
            <div
              class="flex relative"
            >
              <div
                class="absolute flex h-full z-0"
              >
                <input
                  aria-label="Select Newest plant"
                  class="radio my-auto"
                  id=":r2:-26a9fc1f-ef04-4b0f-82ca-f14133fa3b16"
                  name="26a9fc1f-ef04-4b0f-82ca-f14133fa3b16"
                  tabindex="-1"
                  type="checkbox"
                />
              </div>
              <div
                class="w-full overflow-hidden transition-all duration-300 ml-0"
              >
                <a
                  aria-label="Go to Newest plant page"
                  class="collapse cursor-pointer group rounded-2xl bg-neutral text-neutral-content"
                  href="/manage/26a9fc1f-ef04-4b0f-82ca-f14133fa3b16"
                >
                  <input
                    class="hidden pointer-events-none"
                    id=":r5:"
                    type="checkbox"
                  />
                  <div
                    class="collapse-title min-size-0"
                  >
                    <div
                      class="card card-side relative h-24"
                    >
                      <div
                        class="card-body max-w-full my-auto text-center"
                      >
                        <h2
                          class="card-title line-clamp-1 break-words text-center px-8"
                        >
                          Newest plant
                        </h2>
                        <span>
                          Never watered
                        </span>
                      </div>
                      <label
                        aria-label="Show or hide details"
                        class="btn-close absolute right-2 top-8 z-40"
                        for=":r5:"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          class="min-size-8 transition-transform duration-200 rotate-0 group-has-checked:rotate-180"
                          data-slot="icon"
                          fill="currentColor"
                          viewBox="0 0 16 16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M4.22 6.22a.75.75 0 0 1 1.06 0L8 8.94l2.72-2.72a.75.75 0 1 1 1.06 1.06l-3.25 3.25a.75.75 0 0 1-1.06 0L4.22 7.28a.75.75 0 0 1 0-1.06Z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </label>
                    </div>
                  </div>
                  <div
                    class="collapse-content min-w-0"
                  >
                    <div
                      class="pt-4"
                    >
                      <p
                        class="flex"
                      >
                        <span
                          class="font-semibold break-keep mr-4"
                        >
                          Species:
                        </span>
                        <span
                          class="ml-auto text-right line-clamp-1 break-all"
                        >
                          null
                        </span>
                      </p>
                    </div>
                  </div>
                </a>
              </div>
              <label
                class="cursor-pointer absolute z-20"
                for=":r2:-26a9fc1f-ef04-4b0f-82ca-f14133fa3b16"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
    <div
      class="floating-footer floating-footer-hidden"
      data-testid="floating-footer"
    >
      <div
        class="relative flex flex-col items-center gap-4 w-full"
      >
        <div
          class="flex flex-row justify-center gap-8"
        >
          <button
            class="btn btn-neutral w-22"
          >
            Cancel
          </button>
          <button
            class="btn btn-error"
          >
            Remove
          </button>
        </div>
      </div>
    </div>
    <dialog
      class="modal"
    >
      <div
        class="modal-box text-center flex flex-col pt-4 max-w-[25rem]"
      >
        <form
          method="dialog"
        >
          <button
            aria-label="Close modal"
            class="btn-close absolute right-4 top-4"
          >
            <svg
              aria-hidden="true"
              class="size-8 min-size-8"
              data-slot="icon"
              fill="currentColor"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.28 4.22a.75.75 0 0 0-1.06 1.06L6.94 8l-2.72 2.72a.75.75 0 1 0 1.06 1.06L8 9.06l2.72 2.72a.75.75 0 1 0 1.06-1.06L9.06 8l2.72-2.72a.75.75 0 0 0-1.06-1.06L8 6.94 5.28 4.22Z"
              />
            </svg>
          </button>
        </form>
        <h1
          class="font-bold text-lg leading-8 md:text-xl mb-3"
        >
          Edit Details
        </h1>
      </div>
      <form
        class="modal-backdrop"
        method="dialog"
      >
        <button>
          close
        </button>
      </form>
    </dialog>
    <dialog
      class="modal"
    >
      <div
        class="modal-box text-center flex flex-col pt-4 "
      >
        <form
          method="dialog"
        >
          <button
            aria-label="Close modal"
            class="btn-close absolute right-4 top-4"
          >
            <svg
              aria-hidden="true"
              class="size-8 min-size-8"
              data-slot="icon"
              fill="currentColor"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.28 4.22a.75.75 0 0 0-1.06 1.06L6.94 8l-2.72 2.72a.75.75 0 1 0 1.06 1.06L8 9.06l2.72 2.72a.75.75 0 1 0 1.06-1.06L9.06 8l2.72-2.72a.75.75 0 0 0-1.06-1.06L8 6.94 5.28 4.22Z"
              />
            </svg>
          </button>
        </form>
        <h1
          class="font-bold text-lg leading-8 md:text-xl mb-3"
        >
          Add Plants
        </h1>
      </div>
      <form
        class="modal-backdrop"
        method="dialog"
      >
        <button>
          close
        </button>
      </form>
    </dialog>
    <dialog
      class="modal"
    >
      <div
        class="modal-box text-center flex flex-col pt-4 "
      >
        <form
          method="dialog"
        >
          <button
            aria-label="Close modal"
            class="btn-close absolute right-4 top-4"
          >
            <svg
              aria-hidden="true"
              class="size-8 min-size-8"
              data-slot="icon"
              fill="currentColor"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.28 4.22a.75.75 0 0 0-1.06 1.06L6.94 8l-2.72 2.72a.75.75 0 1 0 1.06 1.06L8 9.06l2.72 2.72a.75.75 0 1 0 1.06-1.06L9.06 8l2.72-2.72a.75.75 0 0 0-1.06-1.06L8 6.94 5.28 4.22Z"
              />
            </svg>
          </button>
        </form>
        <h1
          class="font-bold text-lg leading-8 md:text-xl mb-3"
        >
          Change QR Code
        </h1>
      </div>
      <form
        class="modal-backdrop"
        method="dialog"
      >
        <button>
          close
        </button>
      </form>
    </dialog>
  </div>
</div>
`;
